src/main/java/monday/monday/parser/Parser.java

```diff
@@ -0,0 +1,147 @@
+package monday.monday.parser;
+
+import monday.monday.exception.MondayExceptions;
+import monday.monday.ui.Ui;
+import monday.task.Deadline;
+import monday.task.Event;
+import monday.task.TaskList;
+import monday.task.ToDo;
+
+
+/**
+ * Parser class is responsible for parsing user input and performing the corresponding actions.
+ */
+public class Parser {
+    /**
+     * An enumeration of available commands.
+     */
+    private enum Command {
+        BYE,
+        LIST,
+        MARK,
+        UNMARK,
+        TODO,
+        DEADLINE,
+        EVENT,
+        DELETE,
+        FIND
+    }
+
+    /**
+     * Parses the user input and performs the corresponding action.
+     *
+     * @param userInput the user input to be parsed
+     * @param taskList the TaskList object to perform actions on
+     * @return true if the application should continue running, false otherwise
+     * @throws MondayExceptions if there are errors related to the Monday application
+     * @throws IllegalArgumentException if the user input is in the wrong format
+     */
+    public static boolean mondayParser(String userInput, TaskList taskList) throws MondayExceptions {
```

Comment:
Perhaps you could simply rename this method to `parse` instead. Currently, you would have to call `Parser.mondayParser(...)`, which seems a bit verbose. Also, method names should sound like verbs, but it currently sounds like a noun.

Reply:
I agree with you, I'll fix that, thanks for pointing out.